### PR TITLE
always use utf8 encoding to parse json data type columns

### PR DESCRIPTION
**Pull request by panki**
see #409 for details

### CODE REVIEW

File: lib/compile_binary_parser.js [hunk 1]
```diff
@@ -134,8 +134,11 @@ function readCodeFor (field, config, options, fieldNum) {
     return 'packet.readLengthCodedString("ascii");';
   case Types.GEOMETRY:
     return 'packet.parseGeometryValue();';
-  case Types.JSON:
-    return 'JSON.parse(packet.readLengthCodedString(CharsetToEncoding[fields[' + fieldNum + '].characterSet]));';
+    case Types.JSON:
```
Comment (sushantdhiman) [inline lib/compile_binary_parser.js]: ^ Indentation

**Comment by sushantdhiman** [comment]
Tests are failing on Travis as the MySQL Server version used there doesn't support JSON

File: lib/compile_binary_parser.js (hunk 1 above)
Comment (panki) [inline lib/compile_binary_parser.js]: fixed

Comment (sushantdhiman) [inline lib/compile_binary_parser.js]: approved :)

**Comment by panki** [comment]
what should we do with travis?

**Comment by sushantdhiman** [comment]
So here are a few tasks

- Setup MySQL 5.7 on Travis, IIRC they dont support it yet, we will have to use docker
- Windows (Appveyor) is currently unable to detect `eslint` errors in deeply nested folders
- Travis dont fail when `eslint` errors are found

**Comment by sushantdhiman** [comment]
> what should we do with travis?

We should keep the `master` deployable, so unless we setup MySQL 5.7(Travis), this PR will have to wait :(

**Comment by panki** [comment]
@sushantdhiman ok, got it, if i can help you with something, just ket me know

**Comment by sushantdhiman** [comment]
@panki , you can help us by setting up MySQL 5.7 on Travis using Docker. You can continue in this branch as its blocked by it.

Otherwise I'll try to tackle these issues tomorrow :)

**Comment by panki** [comment]
@sushantdhiman of course i can, can you grant me access?

**Comment by sushantdhiman** [comment]
@panki access to what, I dnu if setting up docker will require any additional access ?

**Comment by panki** [comment]
@sushantdhiman sorry, it was stupid question :)

**Comment by panki** [comment]
@sushantdhiman done

**Comment by sidorares** [comment]
Looks good to me! Thanks for setting up dockerized mysql 5.7, this is great.

**Comment by sushantdhiman** [comment]
Thanks for this contribution @panki :+1:

**Pull request merged**